server/src/tasks.ts

```diff
@@ -93,34 +93,26 @@ export class TasksAndModels {
     }
   }
 
-  async addTaskAndModel (task: Task | TaskProvider, model?: tf.LayersModel | URL): Promise<void> {
-    let tfModel: tf.LayersModel
+  async addTaskAndModel (task: Task | TaskProvider, model?: Model | URL): Promise<void> {
     let discoTask: Task
-
     if (isTaskProvider(task)) {
       discoTask = task.getTask()
     } else {
       discoTask = task
     }
 
+    let tfModel: Model
     if (model === undefined) {
       tfModel = await this.loadModelFromTask(task)
-    } else if (model instanceof tf.LayersModel) {
+    } else if (model instanceof Model) {
       tfModel = model
     } else if (model instanceof URL) {
-      tfModel = await tf.loadLayersModel(model.href)
+      tfModel = new models.TFJS(await tf.loadLayersModel(model.href))
```

Comment:
I feel that the `models.TFJS` object is quite confusing, what does it represent? 
If it's only used here can we expose a model builder API rather than exposing the TFJS native constructor? Like `Model.fromURL(...)`

Reply:
Ah I think I understand better, TFJS is the TF.js implementation of the generic abstract Model class. What about renaming it TFjsModel? Or are you relying on the module name `models.TFJS`  to make it clear?

Reply:
> TFJS is the TF.js implementation of the generic abstract Model class.

indeed, the idea is to have multiples implementation of `Model`, one would be wrapping a tfjs model, one would be GPT specific, some other. that allow to keep the old `tf.LayersModel` contained and only providing generic accessor, to better support the coming up models.

> What about renaming it TFjsModel? Or are you relying on the module name `models.TFJS` to make it clear?

I find the indirect import (`models.TFJS`) more explicit, and it's volontarely more cumbersome to write as users should depend on `Model` and not on a specific implementation.